erdos/src/dataflow/stream/extract_stream.rs

```diff
@@ -159,7 +142,7 @@ where
     }
 
     pub fn name(&self) -> String {
-        default_graph::get_stream_name(&self.id)
+        self.name.clone()
```

Comment:
If, as per the previous comment, extract streams should inherit their parent stream name, then we need to retrieve the name from the graph to avoid a consistency issue. Consider the following:
```rust
stream.set_name("s1");
let extract_stream = stream.to_extract_stream(); // Initializes the extract stream with name s1.
stream.set_name("s2");
println!("stream name: {}" stream.name()); // s2
println!("extract_stream name: {}", extract_stream.name()); // s1
```

Reply:
We do indeed have a bigger problem here with the naming. Just as a reference, this PR removes the `set_name` method, because even apart from `ExtractStream`s, the usage of the graph to hold a central point of truth is kind of confusing. Consider, a slightly different example than the one you posted, which would occur if we let developers set names:

```rust
let source_stream = graph.connect_source(...);
source_stream.set_name("MapSourceStream");
let map_stream = graph.connect_one_in_one_out(..., &source_stream);
source_stream.set_name("FilterSourceStream");
let filter_stream = graph.connect_one_in_one_out(..., &source_stream);
````

I'm assuming that in this scenario, if we update names in the graph, that the logs would show that the `MapOperator` is receiving messages on a stream titled `FilterSourceStream`. Is my understanding correct?

Reply:
Yes, that understanding is correct, but in my mind it makes sense. I think there are a few potential solutions:
1. Print a warning (or panic) the 2nd time a stream's name is set.
2. `set_name` returns a `NamedOperatorStream`, which doesn't allow the name to be changed again.

It doesn't look like Flink supports naming streams. ROS has naming built-in with topics.

Reply:
I am more in favor of removing stream names completely, and giving them names according to the operator they are arising out of. `NamedOperatorStream` seems like an abstraction that will just mirror `OperatorStream` but with one method removed.

Reply:
Hmm, I think this can still make debugging difficult. Consider some of the patterns in Pylot, where an operator releases a Union of several types, and these are then split by type. Giving these streams sensible names (rather than something based on the output operator) is very useful for debugging.